### PR TITLE
Added a shutdown feature

### PR DESCRIPTION
**Pull request by maxknivets**
This should do the trick.

### CODE REVIEW

**Comment by shazow** [comment]
While this does give you a way to shut down your server, the goal of this library is to provide FFI bindings, not REST bindings over HTTP. :)

I can't merge this unfortunately.

File: gohttplib.go [hunk 1]
```diff
@@ -23,14 +23,25 @@ import (
...
 	addr := C.GoString(caddr)
-	http.ListenAndServe(addr, nil)
+	srv.Addr = addr
+	Shutdown()
+	srv.ListenAndServe()
+}
+
+// export Shutdown
+func Shutdown() {
+	http.HandleFunc("/shutdown", func(w http.ResponseWriter, r *http.Request) {
```
Comment (shazow) [inline gohttplib.go]: This handler exposes the shutdown over HTTP, which is not something gohttplib should be doing.

Try to set it up so you can call Shutdown() from Python or C.

**Review by shazow — CHANGES_REQUESTED** [review]

File: gohttplib.go (hunk 1 above)
Comment (maxknivets) [reply gohttplib.go]: Alright, so I have re-written the shutdown function so it doesn't work like rest:

https://github.com/maxknivets/gohttplib/blob/master/gohttplib.go#L40

Then I imported the function to python:
https://github.com/maxknivets/gohttplib/blob/master/examples/python/gohttp/__init__.py#L81

And finally, added the function to the list in C file:
https://github.com/maxknivets/gohttplib/blob/master/examples/python/gohttp/build_gohttplib.py#L25

**Comment by shazow** [comment]
That's more like it, good work! :)

Couple of things:
- Please run `make` to make sure all of the generated headers are up to date.
- Can you confirm that this does what you need it to do?

**Comment by maxknivets** [comment]
Alright, I have tested the library so far, and there seems to be an error in the .so file.
The error:
```
$ python3 server1.py
 * Running on http://127.0.0.1:5000/
From cffi callback <function route.<locals>.wrapped.<locals>.handler at 0x7fa54f8c3158>:
Traceback (most recent call last):
  File "/home/max/gohttplib/examples/python/gohttp/__init__.py", line 65, in handler
    fn(ResponseWriter(w), Request(req))
  File "server1.py", line 6, in index
    shutdown()
  File "/home/max/gohttplib/examples/python/gohttp/__init__.py", line 82, in shutdown
    lib.Shutdown()
ffi.error: symbol 'Shutdown' not found in library '/home/max/gohttplib/examples/python/gohttp/libgohttp.so': /home/max/gohttplib/examples/python/gohttp/libgohttp.so: undefined symbol: Shutdown
From cffi callback <function route.<locals>.wrapped.<locals>.handler at 0x7fa54f8c3158>:
Traceback (most recent call last):
  File "/home/max/gohttplib/examples/python/gohttp/__init__.py", line 65, in handler
    fn(ResponseWriter(w), Request(req))
  File "server1.py", line 6, in index
    shutdown()
  File "/home/max/gohttplib/examples/python/gohttp/__init__.py", line 82, in shutdown
    lib.Shutdown()
ffi.error: symbol 'Shutdown' not found in library '/home/max/gohttplib/examples/python/gohttp/libgohttp.so': /home/max/gohttplib/examples/python/gohttp/libgohttp.so: undefined symbol: Shutdown

```
I have updated the [repo](https://github.com/maxknivets/gohttplib) (added the C Shutdown function header). Check it out, there's definitely something I forgot to do.
Also here's the link to the `server1.py` [file](https://gist.github.com/maxknivets/64109f0094c87589fedcbf35206ed3e5)

**Pull request merged**